File: ToDoListApp.Server/Domain/ToDoItem/ToDoItem.cs
namespace ToDoListApp.Server.Domain.ToDoItem
{
    public class ToDoItem
    {
        public int Id { get; private set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public ToDoItemState State { get; private set; } = ToDoItemState.Open;
        public DateTime CreatedAt { get; private set; } = DateTime.UtcNow;
        public DateTime? CompletedAt { get; private set; }

        public void MarkCompleted()
        {
            if (State == ToDoItemState.Completed) return;
            State = ToDoItemState.Completed;
            CompletedAt = DateTime.UtcNow;
        }

        public void Reopen()
        {
            if (State == ToDoItemState.Open) return;
            State = ToDoItemState.Open;
            CompletedAt = null;
        }
    }
}

File: todolistapp.client/src/types/toDoItem.ts
export type ToDoItemState = 'Open' | 'Completed';  

export interface ToDoItemFormData {
    title: string;
    description?: string;
}

export interface ToDoItem {
    id: number;
    title: string;
    description?: string;
    state: ToDoItemState;
    createdAt: string;
    completedAt?: string;
}

export interface CreateToDoItemRequest extends ToDoItemFormData {
}

export interface UpdateToDoItemRequest extends ToDoItemFormData {
    state: ToDoItemState;
}